snapcraft/tests/integration/general/test_metadata_appstream.py

```diff
@@ -76,3 +82,49 @@ def test_specified_metadata_not_overwritten(self):
             snap_yaml['description'], Equals('test-appstream-description'))
         self.assertThat(
             snap_yaml['summary'], Equals('test-summary'))
+
+    def test_icon_extracted_from_appstream(self):
+        os.mkdir('assets')
+        shutil.copyfile(
+            os.path.join(os.path.dirname(tests.__file__), 'data', 'icon.png'),
+            os.path.join(self.path, 'assets', 'test-icon.png'))
+
+        snapcraft_yaml = fixture_setup.SnapcraftYaml(
+            self.path, summary=None, description=None)
+        snapcraft_yaml.data['adopt-info'] = 'test-part'
+        snapcraft_yaml.update_part(
+            'test-part', {
+                'plugin': 'dump',
+                'parse-info': ['test.metainfo.xml']})
+        self.useFixture(snapcraft_yaml)
+
+        self.run_snapcraft('prime')
+
+        self.assertThat(
+            os.path.join('prime', 'meta', 'gui', 'icon.png'), FileExists())
+
+    def test_desktop_extracted_from_appstream(self):
```

Comment:
Can you please add another test for two desktop files?

Reply:
Yes!

Reply:
Hmm did you add it? I don't see it. Seems like you squashed or amended the commit?

Reply:
Aha, nevermind, it's in `test_appstream_with_multiple_launchables`. Thanks a lot for adding that!